examples/nlp/text_classification/conf/config.yaml

```diff
@@ -0,0 +1,81 @@
+# Sentence classification with pretrained BERT models
+
+io:
+  work_dir: ./outputs
+  checkpoint_dir: null
+  save_epoch_freq: 1
+  save_step_freq: -1
+
+pl:
+  trainer:
+    gpus: 1 # the number of gpus, 0 for CPU
+    num_nodes: 1
+    max_epochs: 10
+    max_steps: null # precedence over max_epochs
+    accumulate_grad_batches: 1 # accumulates grads every k batches
+    amp_level: O0 # O1/O2 for mixed precision
+    distributed_backend: ddp
+
+model:
+  data_dir: ??? # /path/to/data
+  class_balancing: null # or weighted_loss
+
+  train_ds:
+    prefix: train
+    batch_size: 32
+    shuffle: true
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
+    drop_last: true
+
+  validation_ds:
+    prefix: dev
+    batch_size: 32
+    shuffle: false
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
+    drop_last: true
+
+  language_model:
+    pretrained_model_name: roberta-base
+    max_seq_length: 36
+    bert_checkpoint: null
+    bert_config: null
+    tokenizer: null # only used if using custom checkpoint
+    tokenizer_model: null # only used if tokenizer is sentencepiece
```

Comment:
tokenizer should be nemobert or sentencepiece. How do we restrict choices with hydra?

Reply:
I think this is how we do that:
```python
@dataclass
class SentencePieceTokenizerConfig:
    name: str = 'sentencepiece'
    model: str = MISSING

@dataclass
class NemoBertTokenizerConfig:
    name: str = 'nemobert'

@dataclass
class LanguageModelConfig:
    pretrained_model_name: Optional[str] = 'roberta-base'
    bert_config: Optional[DictConfig] = None
    max_seq_length: int = 36
    tokenizer: Optional[Union[SentencePieceTokenizerConfig, NemoBertTokenizerConfig]] = NemoBertTokenizerConfig()
```

I think this is a good example that highlights what we can start to do with Hydra and structured configs. Since these tokenizers are common across nlp collections, we can just put them in a nlp collections config object, then all nlp scripts can reuse these same tokenizer configs.

Reply:
Note, sentencepiece wasn't implemented in the script, so this isn't actually being used yet.